nixos/modules/services/networking/pleroma.xml

```diff
@@ -4,129 +4,194 @@
+  </para>
+  <para>Secrets must be moved into a file pointed by <link linkend="opt-services.pleroma.secretConfigFile">services.pleroma.secretConfigFile</link>, in our case <literal>/var/lib/pleroma/secrets.exs</literal>. This file can be created copying the previously generated <literal>config.exs</literal> file and then removing all the settings, except the secrets. This is an example
+<programlisting>
+# Pleroma instance passwords
+
+import Config
+
+config :pleroma, Pleroma.Web.Endpoint,
+   secret_key_base: "U+GAmTd3BTL8Ya9GnBp7fdlH0E9PNgcSOi4jDolPhRAf8SCB+dufEQS1N9WTItpM",
+   signing_salt: "6AsMFwFP"
+
+config :pleroma, Pleroma.Repo,
+  password: "+5dz1GkWNOhTx7CAaDTXMP2OkGc3RYDUi9JDAQoYu+joEdvp+5mhN+aVBuea/R1f",
+
+# Configure web push notifications
+config :web_push_encryption, :vapid_details,
+  public_key: "BL5nHrK9y4Ir8qXvOdZfnulP7_8iTAqgAJt0_ehKX1ga-S2vUyAQoI_xc_3e2z-1cI0JzH-i7bq6_kydER884Aw",
+  private_key: "06tPZ-S0nYdT4-FVyqcFMfBtkkfmrnq0TJarLJxyPlw"
+
+# ... TO CONTINUE ...
+</programlisting>
+  </para>
+  <para>The service can be enabled with the usual
+<programlisting>
+<prompt>$ </prompt>nixos-rebuild switch
+</programlisting>
+  </para>
+  <para>The service is accessible only from the local <literal>127.0.0.1:4000</literal> port. It can be tested using a port forwarding like this
+<programlisting>
+<prompt>$ </prompt>ssh -L 4000:localhost:4000 myuser@example.net
+</programlisting>
+and then accessing <link xlink:href="http://localhost:4000">http://localhost:4000</link> from a web browser.</para>
+ </section>
+ <section xml:id="module-services-pleroma-admin-user">
+  <title>Creating the admin user</title>
+  <para>After Pleroma service is running, all <link xlink:href="https://docs-develop.pleroma.social/">Pleroma administration utilities</link> can be used. In particular an admin user can be created with
+<programlisting>
+<prompt>$ </prompt>pleroma_ctl user new &lt;nickname&gt; &lt;email&gt;  --admin --moderator --password &lt;password&gt;
+</programlisting>
+  </para>
+ </section>
+ <section xml:id="module-services-pleroma-nginx">
+  <title>Configuring Nginx</title>
+  <para>In this configuration, Pleroma is listening only on the local port 4000. Nginx can be configured as a Reverse Proxy, for forwarding requests from public ports to the Pleroma service. This is an example of configuration, using
+<link xlink:href="https://letsencrypt.org/">Let's Encrypt</link> for the TLS certificates
+<programlisting>
+security.acme = {
+  email = "root@example.net";
+  acceptTerms = true;
+};
+
+services.nginx = {
+  enable = true;
+  addSSL = true;
+
+  recommendedProxySettings = false;
+  # NOTE: if enabled, the Proxy optimizations will override the Pleroma
+  # specific settings, and they will enter in conflict.
+
+  virtualHosts = {
+    # Used for ACME certificate validations
+
+    "acmechallenge.example.net" = {
+      # Catchall vhost, will redirect users to HTTPS for all vhosts
+      serverAliases = [ "*.example.net" ];
```

Comment:
Meh. Pleroma only requires a single vhost to operate. I don't see why you'd require a wildcard certificate just for this service.

Let's encourage users to use good security practices and only issue the certificate for the domain name we actually need.

Reply:
Ops! I miss interpreted the section of NixOS manual about ACME/Let's Encrypt https://nixos.org/manual/nixos/stable/index.html#module-security-acme-configuring . I thought this setting was mandatory. Instead it suffices a lot less!!!

I fixed this on my server, and tested creating new domains and it works. So I committed a simpler and safer version.